File: hackerearth/sorting/bubble_sort.cpp
#include <iostream>
#include <vector>

using namespace std;

int bubbleSort(vector<int> a)
{
    int count = 0;
    int n = a.size();
    bool swapped = true;

    while (swapped)
    {
        swapped = false;
        count = count+1;
        for (int i = 0; i < n-2; i++)
        {
            if (a[i] > a[i+1])
            {
                int temp = a[i];
                a[i] = a[i+1];
                a[i+1] = temp;
                swapped = true;
            }
        }
    }

    return count;

}

int main() {
    int num;
    cin >> num;  
    vector<int> arr;
    while (num --)
    {
        int x;
        cin >> x;

        arr.push_back(x);

    }

    cout << bubbleSort(arr);

}

File: algorithm_py/bubble_sort.py
arr = [10, 9, 8, 7, 6, 5, 4, 3, 2, 1]

for i in range(len(arr)):
    for j in range(len(arr)):
        if j+1 < len(arr) and arr[j] > arr[j+1]:
            (arr[j], arr[j+1]) = (arr[j+1], arr[j])

print(arr)
